Clarify terms, query intervals and impacts (#469)
owncloud/docs-client-desktop: modules/ROOT/pages/faq.adoc

{description}

== Clarification of Terms Used

* *Polling* +

> Polling is defined as the task where the Desktop app queries the server for changes. If the server responds with a change that had happened on the backend, the client will start syncing the changes after polling has finished.

Reviewer: Again: we poll a _lot_ of information. Polling is _not_ limited to this singular use case you describe here. This wording is misleading, as it suggests that this is the one and only time we use the term polling to describe a task.

> Polling is defined as the task where the Desktop app queries the server for changes. If the server responds with a change that had happened on the backend, the client will start syncing the changes after polling has finished.

Author: Thanks, will commit, a good addon also covering @fmoc comment.

> Polling is defined as the task where the Desktop app queries the server for changes. If the server responds with a change that had happened on the backend, the client will start syncing the changes after polling has finished. If there is a proxy configured, it can happen that the proxy does not return actual changes but a cached state.

Author: Though the proxy is only one possibe cause, we need to give at least a hint. Else we are at the same state as before not knowing the reasons for the various options/settings. Keeping it as it was written.

> Polling is defined as the task where the Desktop app queries the server for changes. If the server responds with a change that had happened on the backend, the client will start syncing the changes after polling has finished. If there is a proxy configured, it can happen that the proxy does not return actual changes but a cached state.

Reviewer 2: ```suggestion
Polling is defined as the task where the Desktop app queries the server for changes. If the server responds with a change that had happened on the backend, the client will start syncing the changes after polling has finished.
```

Sounds like an error. Where does this come from? I can't remember this ever happened.

> Polling is defined as the task where the Desktop app queries the server for changes. If the server responds with a change that had happend on the backend, the client will start syncing the changes after polling has finished. Note that when there is a proxy configured, it can happen that the proxy does not return actual changes but a cached state.

Reviewer 3: ```suggestion
Polling is defined as the task where the Desktop app queries the server for changes. If the server responds with a change that had happened on the backend, the client will start syncing the changes after polling has finished. If there is a proxy configured, it can happen that the proxy does not return actual changes but a cached state.
```

> * *Syncing* +

Reviewer: ```suggestion
* *Sync* +
```

> Syncing is defined as the task where any changes will be synchronized, no matter if they occurred in the Desktop app or on the server side. Syncing does not depend on former polling while polling can result in a sync.

Reviewer: "Syncing" is not used as a term anywhere in the code or docs, while "sync" is. We should avoid using the present participle and treat "sync" as a specific noun.

Reviewer 3: Linguistically there's nothing wrong with syncing as the gerund (noun) or continuous form of verbs. In the documentation we use it a lot. No reason to ban certain forms of word.
https://www.merriam-webster.com/dictionary/syncing

> Syncing is defined as the task where any changes will be synchronized, no matter if they occurred in the Desktop app or on the server side. If syncing is enforced, it does not depend on former polling.

Author: ```suggestion
Syncing is defined as the task where any changes will be synchronized, no matter if they occurred in the Desktop app or on the server side. Syncing does not depend on former polling while polling can result in a sync.
```

> Syncing is defined as the task where any changes will be synchronized, no matter if they occurred in the Desktop app or on the service side. If syncing is enforced, it does not depend on former polling.

Reviewer: ```suggestion
Every time the Desktop app needs to synchronize changes between its host and the server, a so-called sync is run. In such a sync, server-side changes are applied locally and local changes are sent to the server. In the end, both sides are in sync with each other.
```

A reference to polling is misleading. Polling just describes the query part. A polling request may lead to a sync, but they are distinct from each other.

Author: We must make the terms distinguishable and why/why not terms relate. I will relax the statement to make it more clear.

> Syncing is defined as the task where any changes will be synchronized, no matter if they occurred in the Desktop app or on the service side. If syncing is enforced, it does not depend on former polling.

Reviewer 4: ```suggestion
Syncing is defined as the task where any changes will be synchronized, no matter if they occurred in the Desktop app or on the server side. If syncing is enforced, it does not depend on former polling.
```

> Syncing is defined as task where any changes, independent if occured at the Desktop app or on the service side will be synchronized. If syncing is enforced, it is not dependent on former polling.

Reviewer 3: ```suggestion
Syncing is defined as the task where any changes will be synchronized, no matter if they occurred in the Desktop app or on the service side. If syncing is enforced, it does not depend on former polling.
```

* *Local Discovery* +

> Local discovery is defined as task where the Desktop app looks for changes on the local filesystem. A local discovery can internally be enforced to check for changes that have been missed like when the Desktop app was shut down during changes happened on the local filesystem. Any changes identified on the local filesystem trigger a sync after local discovery has finished.

Reviewer 3: ```suggestion
Local discovery is defined as the task where the Desktop app looks for changes on the local filesystem. A local discovery can internally be enforced to check for changes that have been missed for example when the Desktop app was shut down during ongoing changes on the local filesystem. Any changes identified on the local filesystem trigger a sync after local discovery has finished.
```

> Local discovery is defined as the task where the Desktop app looks for changes on the local filesystem. A local discovery can internally be enforced to check for changes that have been missed for example when the Desktop app was shut down during ongoing changes on the local filesystem. Any changes identified on the local filesystem trigger a sync after local discovery has finished.

Reviewer: ```suggestion
During local discovery, the Desktop app actively looks for changes on the local filesystem. Should there be any, the Desktop app will commence a sync once it has finished the discovery. Local discovery is needed for instance when the Desktop app was just started to make sure changes made while it was shut down are synchronized with the server. Furthermore, local discovery is enforced in regular intervals to catch changes the operating system has not forwarded correctly to the Desktop app.
```

Reviewer: I am not a fan of the construct "is defined as the task where", because there are more elegant wordings. Please do not hesitate to merge my suggestion with your description.
